dear_petition/petition/models.py

```diff
@@ -52,52 +113,73 @@ def parse_report(self, report_pdf=None):
                 logger.exception(e)
                 data = {"error": str(e)}
             return data
-
-    @property
-    def file_no(self):
-        return self.data["General"].get("File No", "")
-
-    @property
-    def county(self):
-        return self.data["General"].get("County", "")
-
-    @property
-    def dob(self):
-        return self.data["Defendant"].get("Date of Birth/Estimated Age", "")
-
-    @property
-    def case_status(self):
-        return self.data["Case Information"].get("Case Status", "")
-
-    @property
-    def offense_date(self):
-        return self.data["Case Information"].get("Offense Date", "")
-
-    @property
-    def arrest_date(self):
-        return self.data["Offense Record"].get("Arrest Date", self.offense_date)
-
-    @property
-    def disposed_on(self):
-        return self.data["Offense Record"].get("Disposed On", "")
-
-    @property
-    def disposition_method(self):
-        return self.data["Offense Record"].get("Disposition Method", "")
-
-    @property
-    def district_court(self):
-        return self.data["General"].get("District", "")
-
-    @property
-    def superior_court(self):
-        return self.data["General"].get("Superior", "")
-
-    @property
-    def offenses(self):
-        for offense in self.data["Offense Record"].get("Records", []):
-            yield offense
-
+    
+    def refresh_record(self):
```

Comment:
This looks cool! One idea would be to consolidate the logic into the refresh() funciton, so that both the manager and the migration could use it. I haven't tried running this code yet, but maybe something like this:

```diff
diff --git a/dear_petition/petition/models.py b/dear_petition/petition/models.py
index d3ec5e8..6cb8576 100644
--- a/dear_petition/petition/models.py
+++ b/dear_petition/petition/models.py
@@ -23,47 +23,24 @@ from .constants import (
     SEX_CHOICES,
     MALE,
     FEMALE,
-    UNISEX
+    UNISEX,
 )
 
 
 logger = logging.getLogger(__name__)
 
-class CIPRSRecordManager(models.Manager):
 
-    def create_record(self, batch, date_uploaded, report_pdf, label, data):
-        '''Extract General, Case, and Defendant details from data 
+class CIPRSRecordManager(models.Manager):
+    def create_record(self, data):
+        """Extract General, Case, and Defendant details from data
 
         Parses the raw data from our JSONField (data) and
         places values in their associated model fields
-        '''
-        file_no = self.get_file_no(data)
-        county = self.get_county(data)
-        dob = self.get_dob(data)
-        sex = self.get_sex(data)
-        race = self.get_race(data)
-        case_status = self.get_case_status(data)
-        offense_date = self.get_offense_date(data)
-        arrest_date = self.get_arrest_date(data)
-        jurisdiction = self.get_jurisdiction(data)
-
-        ciprs_record = self.create(
-            batch=batch,
-            date_uploaded=date_uploaded,
-            report_pdf=report_pdf,
-            label=label,
-            data=data,
-            file_no=file_no,
-            county=county,
-            dob=dob,
-            sex=sex,
-            race=race,
-            case_status=case_status,
-            offense_date=offense_date,
-            arrest_date=arrest_date,
-            jurisdiction=jurisdiction
-        )
-        return ciprs_record
+        """
+        ciprs_record = CIPRSRecord(data)
+        ciprs_record.refresh_record_from_data()
+        ciprs_record.save()
+
 
 class CIPRSRecord(models.Model):
 
@@ -83,9 +60,7 @@ class CIPRSRecord(models.Model):
     offense_date = models.DateTimeField(null=True)
     arrest_date = models.DateField(null=True)
     jurisdiction = models.CharField(
-        max_length=16,
-        choices=JURISDICTION_CHOICES,
-        default=NOT_AVAILABLE
+        max_length=16, choices=JURISDICTION_CHOICES, default=NOT_AVAILABLE
     )
 
     objects = CIPRSRecordManager()
@@ -113,73 +88,25 @@ class CIPRSRecord(models.Model):
                 logger.exception(e)
                 data = {"error": str(e)}
             return data
-    
-    def refresh_record(self):
-        '''Updates model fields that depends on the data JSONField
+
+    def refresh_record_from_data(self):
+        """Updates model fields that depends on the data JSONField
 
         The record exists, but the raw data (data - JSONFIELD) has
         changed. Let's update the models that are extracting data
         from this field.
-        '''
-        record = CIPRSRecord.objects.get(pk=self.pk)
-        record.file_no = self.get_file_no(self.data)
-        record.county = self.get_county(self.data)
-        record.dob = self.get_dob(self.data)
-        record.sex = self.get_sex(self.data)
-        record.race = self.get_race(self.data)
-        record.case_status = self.get_case_status(self.data)
-        record.offense_date = self.get_offense_date(self.data)
-        record.arrest_date = self.get_arrest_date(self.data)
-        record.jurisdiction = self.get_jurisdiction(self.data)
-        
-        
-        return record.save(update_fields=[
-            'file_no',
-            'county',
-            'dob',
-            'sex',
-            'race',
-            'case_status',
-            'offense_date',
-            'arrest_date',
-            'jurisdiction'
-        ])
-
-    def get_file_no(self, data):
-        return data["General"].get("File No", "")
-
-    def get_county(self, data):
-        return data["General"].get("County", "")
-
-    def get_dob(self, data):
-        return data["Defendant"].get("Date of Birth/Estimated Age", "")
-    
-    def get_sex(self, data):
-        return data["Defendant"].get("Sex", "")
-
-    def get_race(self, data):
-        return data["Defendant"].get("Race", "")
-
-    def get_case_status(self, data):
-        return data["Case Information"].get("Case Status", "")
-
-    def get_offense_date(self, data):
-        return data["Case Information"].get("Offense Date", "")
-
-    def get_jurisdiction(self, data):
-        is_superior = data["General"].get("Superior", "")
-        is_district = data["General"].get("District", "")
-        if is_superior:
-            return SUPERIOR_COURT
-        elif is_district:
-            return DISTRICT_COURT
-        else:
-            return NOT_AVAILABLE
-
-    def get_arrest_date(self, data):
-        offense_date = data["Case Information"].get("Offense Date", "")
-        return data["Offense Record"].get("Arrest Date", offense_date)
- 
+        """
+        self.file_no = self.data["General"].get("File No", "")
+        self.county = self.data["General"].get("County", "")
+        self.dob = self.data["Defendant"].get("Date of Birth/Estimated Age", "")
+        self.sex = self.data["Defendant"].get("Sex", "")
+        self.race = self.data["Defendant"].get("Race", "")
+        self.case_status = self.data["Case Information"].get("Case Status", "")
+        self.offense_date = self.data["Case Information"].get("Offense Date", "")
+        self.arrest_date = self.data["Offense Record"].get("Arrest Date", "")
+        self.jurisdiction = ""  # todo
+        self.save()
+
 
 class Contact(models.Model):
 
```

Reply:
I made the above change to `refresh_record_from_data`, but I though wanted to keep the methods (that were @properties) as "legacy" methods. This may not be necessary, but I remember talking about this.

Reply:
Also, we removed `batch`, `date_uploaded`, `report_pdf`, and `label` from `create_record`. But since we actually creating a record through this `queryset` method I think we still need to pass these parameters to `CIPRSRecord()`. What do you all think?

Reply:
Thoughts:

1. legacy methods:

I'm ok with keeping the legacy methods or not; I'm not sure which is more helpful for the project.

2. parameters to `create_record()`:
Good point about passing `batch`, `date_uploaded`, `report_pdf`, and `label` to `create_record()`. Yeah, I think it would be easiest to pass those as parameters to `create_record()`, and set them like
```
ciprs_record = CIPRSRecord(data)
ciprs_record.batch = batch
ciprs_record.date_uploaded = date_uploaded
ciprs_record.ciprs_record.date_uploaded = date_uploaded
ciprs_record.label = label
ciprs_record.refresh_record_from_data()
ciprs_record.save()
```

I'll let @copelco share thoughts too, though, since he knows more about the project

Reply:
I do not think methods on a model a part of the `Historical Model` (they cannot be used during migrations).

Reply:
Also, I'll post Colin's comment from Slack here: " i think the goal should be to move away from those in the end. some code currently uses them, so those references will need to be updated before they're official removed though."
I'm ok with removing them, and fixing the references